template/interpolate/funcs.go

```diff
@@ -196,7 +196,23 @@ func funcGenBuild(ctx *Context) interface{} {
 			}
 			return "", fmt.Errorf("loaded data, but couldnt find %s in it.", s)
 		}
-
+		if data, ok := ctx.Data.(map[string]interface{}); ok {
+			// PlaceholderData has been passed into generator, so if the given
```

Comment:
What do you think about updating these if statements into a type switch? It will handle the ok checks for us and reduce the number of if statements.